dns/zone-configs/k8s.io.yaml

```diff
@@ -232,3 +228,76 @@ velodrome:
 kind.sigs:
   type: CNAME
   value: k8s-kind.netlify.com.
+
+# TODO: Split into separate file
+# URL shortener service.
+go:
+  type: CNAME
+  value: txtd.io.
```

Comment:
I am very uncomfortable taking such a dependency on a free service that has no documented SLA, no public monitoring, and no repercussions to an outage.

I'd rather run something ourselves.

If we think this is a cleaner model for config (I have no strong feeling - @misterikkit ?) it seems pretty trivial to write a simple form of this that we can make our own guarantees about.

Reply:
@thockin It's an OSS project: https://github.com/txtdirect/txtdirect

We could easily create a manifest and run it ourselves, but I think @stp-ip was just POC'ing it out.

Reply:
Basically my team and I run the "free" service. It's doing around 2 million requests per month and has monitoring etc.
But this was suggested as a first step to test on a canary beforehand and then move to a self-hosted option.
We can provide the k8s deployment code (as that's what we also run the hosted service on) and publicly build docker images for txtdirect: https://gcr.io/v2/txtdirect-223710/txtdirect/tags/list

As @cblecker mentioned this is mostly a PoC to get the process and feeling and then be able to switch to self-hosted for prod.

Reply:
As said. All open source and definitely not pushing the hosted option. In my mind it always was supposed to run the self-hosted after the PoC.

Reply:
Roger.  Sorry for being harsh.  It has nothing to do with YOUR service and everything to do with A service. :)

Reply:
Regarding the config model, this is more complex than my POC. It puts structured data in the TXT record and that record needs to go in a DNS config file. People will likely be copy-pasting existing records when adding new ones. (that's probably fine)

a potential benefit is that our DNS config is well maintained, versus a file of custom schema in a different part of the repo.

The main reason I've come around to the side of using TXTDirect is the operational complexity is low compared to what I was building before.

Reply:
To help with people understanding the records more easily and making copy-pasting easier. I moved all "static" config of the record to the start and the last value is the one for `to=` and hopefully can be easily grasped and changed.

Reply:
> Roger. Sorry for being harsh. It has nothing to do with YOUR service and everything to do with A service. :)

No worries. Just wanted to give context and as mentioned never was the intention to add A service dependency.